src/bin/inx-chronicle/cli.rs

```diff
@@ -33,4 +38,97 @@ pub struct ClArgs {
     /// Toggle the metrics server.
     #[clap(long, value_parser, env = "METRICS")]
     pub enable_metrics: Option<bool>,
+    /// Subcommands.
+    #[clap(subcommand)]
+    pub subcommand: Option<Subcommands>,
+}
+
+impl ClArgs {
+    /// Get a config file with CLI args applied.
+    pub fn get_config(&self) -> Result<ChronicleConfig, ConfigError> {
+        let mut config = self
+            .config
+            .as_ref()
+            .map(ChronicleConfig::from_file)
+            .transpose()?
+            .unwrap_or_default();
+
+        if let Some(connect_url) = &self.mongodb_conn_url {
+            config.mongodb.connect_url = connect_url.clone();
+        }
+
+        #[cfg(all(feature = "stardust", feature = "inx"))]
+        {
+            if let Some(connect_url) = &self.inx_addr {
+                config.inx.connect_url = connect_url.clone();
+            }
+            if let Some(enabled) = self.enable_inx {
+                config.inx.enabled = enabled;
+            }
+        }
+
+        #[cfg(feature = "api")]
+        {
+            if let Some(password) = &self.password {
+                config.api.password_hash = hex::encode(
+                    argon2::hash_raw(
+                        password.as_bytes(),
+                        config.api.password_salt.as_bytes(),
+                        &Into::into(&config.api.argon_config),
+                    )
+                    .expect("invalid JWT config"),
+                );
+            }
+            if let Some(path) = &self.identity_path {
+                config.api.identity_path.replace(path.clone());
+            }
+            if let Some(enabled) = self.enable_api {
+                config.api.enabled = enabled;
+            }
+        }
+
+        if let Some(enabled) = self.enable_metrics {
+            config.metrics.enabled = enabled;
+        }
+
+        Ok(config)
+    }
+
+    /// Process subcommands and return whether the app should early exit.
+    #[allow(unused)]
+    #[allow(clippy::collapsible_match)]
+    pub fn process_subcommands(&self, config: &ChronicleConfig) -> bool {
```

Comment:
Shouldn't this return an error type?

Reply:
I don't think it's necessary for CLI operations.

Reply:
What if there was an error during any of the subcommands?

Reply:
Then it will panic 😆

Reply:
Panics should be reserved for things that we didn't anticipate to happen. Since some of these commands can very well return errors we should have a mechanism to catch them and shut down things properly. This will also be required once we have more administrative functionality in the CLI.

Reply:
This doesn't start anything up to shut down.